### PR TITLE
New view competitions

### PR DESCRIPTION
**Pull request by mthembbu**

### CODE REVIEW

File: skietbaan_fe/src/App.js [hunk 1]
```diff
@@ -47,21 +47,21 @@ class App extends Component {
 								<Route path="/AddGroup" component={AddGroup} exact />
 								<Route path="/scoreCapture" component={ScoreCapture} exact />
 								<Route path="/groups" component={Groups} exact />
-								<Route path="/viewComp" component={ViewComp} exact />
-								<Route path="/createComp" component={CreateComp} exact />
+
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/App.js]: remove empty lines

File: skietbaan_fe/src/App.js [hunk 2]
```diff
@@ -47,21 +47,21 @@ class App extends Component {
...
 								<Route path="/notify" component={notification} exact />
 								<Route path="/documents" component={Documents} exact />
 								<Route path="/profile" component={ProfileLanding} exact />
 								<Route path="/requirements" component={requirements} exact />
 								<Route path={'/ViewGroups'} component={GroupComponent} exact />
-								<Route path="/create" component={createPages} exact />
+								<Route path={'/GroupComponent'} component={GroupComponent} exact />
+								<Route path={'/create'} component={createPages} exact />
+
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/App.js]: Ditto

File: skietbaan_fe/src/actions/types.js [hunk 3]
```diff
@@ -43,7 +45,8 @@ export const UPDATE_SELECTED_COMPETITION = "UPDATE_SELECTED_COMPETITION";
 export const UPDATE_SELECTED_GROUP = "UPDATE_SELECTED_GROUP";
 
 //groups types and constants
-export const URL = "https://api.skietbaan.retrotest.co.za";
+//https://api.skietbaan.retrotest.co.za"
+export const URL = "http://localhost:50859";
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/actions/types.js]: should this still point to localhost when it's going into dev?

File: skietbaan_fe/src/components/CreateComp.js [hunk 4]
```diff
@@ -46,22 +47,27 @@ class CreateComp extends Component {
...
-        }
+		}
 		return (
 			<div>
-				
 				<div class="create-comp-container">
 					<Form onSubmit={this.onSubmit}>
 						<div className="containers-input">
-							<label className="comp-label-container">Competition Name</label>
+							<label className="comp-label-container">{/** Competition Name*/}</label>
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/components/CreateComp.js]: Is this label required when it's commented?

File: skietbaan_fe/src/components/CreateComp.js [hunk 5]
```diff
@@ -46,22 +47,27 @@ class CreateComp extends Component {
...
+			Hours: this.state.Hours,
 			Status: true
 		};
 		this.props.createcomp(compData);
+		if(this.props.isExist == false)
+		{
+			history.push("/ViewComp");
+		}
 		this.setState({ isExist: this.props.isExist });
+		console.log('Inside the onSubmit isExist: ',this.props.isExist);
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/components/CreateComp.js]: Remove

Comment (kyle-manganyi) [inline skietbaan_fe/src/components/CreateComp.js]: do you need this console log?

File: skietbaan_fe/src/actions/types.js (hunk 3 above)
Comment (mthembbu) [reply skietbaan_fe/src/actions/types.js]: resolved

File: skietbaan_fe/src/actions/competition.action.js [hunk 6]
```diff
@@ -1,82 +1,107 @@
...
+	fetch(URL + '/R/' + CompID)
+		.then((response) => response.json())
+		.then((requirementsData) => {
+			dispatch({
+				type: FETCH_REQ,
+				payload: requirementsData
+			});
+		})
+		.catch((error) => {
+			console.log('Error Message: ', error);
```
Comment (ilzeLourens) [inline skietbaan_fe/src/actions/competition.action.js]: remove console

File: skietbaan_fe/src/actions/competition.action.js [hunk 7]
```diff
@@ -1,82 +1,107 @@
...
+		method: 'POST',
+		headers: {
+			Accept: 'application/json',
+			'Content-Type': 'application/json'
+		},
+		body: JSON.stringify(compData)
+	})
+		.then((res) => {
+			if (res.ok) {
+				console.log('Response status in an If(): ', res.statusText);
```
Comment (ilzeLourens) [inline skietbaan_fe/src/actions/competition.action.js]: console log

File: skietbaan_fe/src/actions/competition.action.js [hunk 8]
```diff
@@ -1,82 +1,107 @@
...
+			} else {
+				res.json().then(() => {
+					dispatch({
+						type: NEW_COMP,
+						payload: true
+					});
+				});
+			}
+			console.log('Final returned, Response Status: ', res.ok);
+		})
```
Comment (ilzeLourens) [inline skietbaan_fe/src/actions/competition.action.js]: another

File: skietbaan_fe/src/actions/competition.action.js [hunk 9]
```diff
@@ -1,82 +1,107 @@
...
+		.catch((error) => {
+			console.log('request error => ', error);
```
Comment (ilzeLourens) [inline skietbaan_fe/src/actions/competition.action.js]: console ;log

File: skietbaan_fe/src/actions/competition.action.js [hunk 10]
```diff
@@ -1,82 +1,107 @@
...
+				);
+			} else {
+				res.json().then(() => {
+					dispatch({
+						type: NEW_COMP,
+						payload: true
+					});
+				});
+			}
+			console.log('Final returned, Response Status: ', res.ok);
```
Comment (kyle-manganyi) [inline skietbaan_fe/src/actions/competition.action.js]: do you need this console log?

File: skietbaan_fe/src/actions/types.js [hunk 11]
```diff
@@ -43,6 +45,7 @@ export const UPDATE_SELECTED_COMPETITION = "UPDATE_SELECTED_COMPETITION";
 export const UPDATE_SELECTED_GROUP = "UPDATE_SELECTED_GROUP";
 
 //groups types and constants
+//https://api.skietbaan.retrotest.co.za"
```
Comment (ilzeLourens) [inline skietbaan_fe/src/actions/types.js]: GC

Comment (kyle-manganyi) [inline skietbaan_fe/src/actions/types.js]: GC

File: skietbaan_fe/src/components/CreateComp.js [hunk 12]
```diff
@@ -46,22 +47,26 @@ class CreateComp extends Component {
 		const compData = {
 			Name: this.state.Name /** Don't forget to change to lowercase to avoid case sensitivity conflicts*/,
```
Comment (ilzeLourens) [inline skietbaan_fe/src/components/CreateComp.js]: Did you do this

File: skietbaan_fe/src/scss/view-comp.css [hunk 13]
```diff
@@ -78,12 +110,78 @@
 }
 
 .td-col{
+    vertical-align: top;
 	padding-top: 11px;
 	padding-bottom: 11px;
-	text-align: left;
+    width: 90%;/*don't forget to remove*/
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/scss/view-comp.css]: What does this mean?

File: skietbaan_fe/src/scss/view-comp.css [hunk 14]
```diff
@@ -78,12 +110,78 @@
...
+}
+
+
+.comp-req-container {
+	margin: 20px;
+	padding: 20px;
+	text-align: center;
+}
+
+.req-input {
```
Comment (kyle-manganyi) [inline skietbaan_fe/src/scss/view-comp.css]: try using prettier so code is formatted properly, seems like most spacing is inconsistent

File: skietbaan_fe/src/actions/types.js (hunk 11 above)
Comment (mthembbu) [reply skietbaan_fe/src/actions/types.js]: sorted

File: skietbaan_fe/src/components/ViewComp.js [hunk 15]
```diff
@@ -1,130 +1,363 @@
...
+	//End-point: URL/R/{ID}
+	getRequirements(index) {
+		fetch(URL + '/R/' + index).then((response) => response.json()).then((requirementsData) =>{
+			this.setState({
+				bronzeAccuracy: requirementsData[0].accuracy,
+				bronzeTotal: requirementsData[0].total,
+				silverAccuracy: requirementsData[1].accuracy,
+				silverTotal: requirementsData[1].total,
+				goldAccuracy: requirementsData[1].accuracy,
+				goldTotal: requirementsData[2].total
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/components/ViewComp.js]: Should this not be goldAccuracy: requirementsData[2].accuracy

File: skietbaan_fe/src/scss/view-comp.css (hunk 14 above)
Comment (JurieOberholzer) [reply skietbaan_fe/src/scss/view-comp.css]: Yes, or just indent like normally.

File: skietbaan_fe/src/components/ViewComp.js (hunk 15 above)
Comment (mthembbu) [reply skietbaan_fe/src/components/ViewComp.js]: Thanks, resolved. Wii PR just now

File: skietbaan_fe/src/components/CreateComp.js [hunk 16]
```diff
@@ -46,22 +47,26 @@ class CreateComp extends Component {
...
 		this.props.createcomp(compData);
+		if(this.props.isExist == false)
+		{
+			history.push("/ViewComp");
+		}
 		this.setState({ isExist: this.props.isExist });
 	}
 	render() {
 		if(!getCookie("token")){
             window.location = "/registerPage";
```
Comment (JurieOberholzer) [inline skietbaan_fe/src/components/CreateComp.js]: Remove if statement for getcookie en remove import

File: skietbaan_fe/src/components/ViewComp.js [hunk 17]
```diff
@@ -10,7 +10,7 @@ import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import '../scss/view-comp.css';
 import { URL } from '../actions/types';
-import Collapsible from 'react-collapsible';
+import letterhead from './compassets/letterofgoodstanding.png';
```
Comment (ilzeLourens) [inline skietbaan_fe/src/components/ViewComp.js]: camelCase

**Pull request closed without merging**